bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/api/rest/FreeboxOsSession.java

```diff
@@ -111,7 +114,7 @@ public void closeSession() {
         return localSession != null ? localSession.getSessionToken() : null;
     }
 
-    private <F, T extends Response<F>> @Nullable F execute(URI uri, HttpMethod method, Class<T> clazz,
+    private synchronized <F, T extends Response<F>> @Nullable F execute(URI uri, HttpMethod method, Class<T> clazz,
```

Comment:
Synchronized method in case the session is expired. If two session is open in same time, the first session become invalid, so if this method is not synchronized, le query may fail.